### PR TITLE
fix: decode C-quoted paths from git diff

### PR DESCRIPTION
**Pull request by vitali87**
## Summary

Even with `core.quotePath=false` (#93), git C-quotes any path containing `"`, `\` or a control character:

```
diff --git "a/we\"ird.py" "b/we\"ird.py"
```

unidiff keeps the quotes and escapes verbatim, so `pf.path` came out as the literal string `"b/we\"ird.py"` — prefix not stripped, escapes intact — and the worker would have written the sub-PR file under that name.

`parse_diff` now decodes `source_file`/`target_file` with a new `unquote_git_path` (C escapes plus octal bytes, reassembled as UTF-8), so every downstream consumer (`git show base:path`, `target_file_modes`, chunk headers, the worker) sees the real path.

Stacked on #93 (`fix/unicode-paths-in-diff`).

## Test plan

- parametrized decoder cases (`\"`, `\t`, `\\`, octal UTF-8, `\a`/`\1`, edge cases), a parsed-headers test with added and deleted quoted files, and a real-`git diff` e2e test with `"`, tab and backslash in names — all fail on the base parser
- 474 tests pass, ruff clean
- Local review: 5/5

Known follow-up (pre-existing, not addressed here): unidiff mis-splits `diff --git` lines when a *quoted* path also contains a space.

### CODE REVIEW

File: pr_split/diff_ops/parser.py [hunk 1]
```diff
@@ -52,11 +53,55 @@ def extract_diff(dev_branch: str, base_branch: str) -> str:
...
+        out += body[pos : match.start()].encode("utf-8")
+        escape = match.group(1)
+        if escape[0] in "01234567":
+            # Octal escapes are single bytes; consecutive ones form one
+            # multi-byte UTF-8 character, which the final decode reassembles.
+            out.append(int(escape, 8))
+        else:
+            out += _C_ESCAPES.get(escape, escape.encode("utf-8"))
+        pos = match.end()
+    out += body[pos:].encode("utf-8")
```
[the remark this answers is not in the document]
Comment (vitali87) [reply pr_split/diff_ops/parser.py]: Fixed: literal spans are now encoded with surrogateescape, so a quoted path holding a non-UTF-8 byte round-trips. Covered by a new TestUnquoteGitPath case.
